homeassistant/components/switcher_kis/climate.py

```diff
@@ -0,0 +1,240 @@
+    SwitcherType2Api,
+)
+from aioswitcher.device import (
+    DeviceCategory,
+    DeviceState,
+    SwitcherThermostat,
+    ThermostatFanLevel,
+    ThermostatMode,
+    ThermostatSwing,
+)
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    SWING_OFF,
+    SWING_VERTICAL,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitcherDataUpdateCoordinator
+from .const import SIGNAL_DEVICE_ADD
+
+DEVICE_MODE_TO_HA = {
+    ThermostatMode.COOL: HVACMode.COOL,
+    ThermostatMode.HEAT: HVACMode.HEAT,
+    ThermostatMode.FAN: HVACMode.FAN_ONLY,
+    ThermostatMode.DRY: HVACMode.DRY,
+    ThermostatMode.AUTO: HVACMode.HEAT_COOL,
+}
+
+HA_TO_DEVICE_MODE = {value: key for key, value in DEVICE_MODE_TO_HA.items()}
+
+DEVICE_FAN_TO_HA = {
+    ThermostatFanLevel.LOW: FAN_LOW,
+    ThermostatFanLevel.MEDIUM: FAN_MEDIUM,
+    ThermostatFanLevel.HIGH: FAN_HIGH,
+    ThermostatFanLevel.AUTO: FAN_AUTO,
+}
+
+HA_TO_DEVICE_FAN = {value: key for key, value in DEVICE_FAN_TO_HA.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Switcher climate from config entry."""
+    remote_manager = BreezeRemoteManager()
+
+    async def async_add_climate(coordinator: SwitcherDataUpdateCoordinator) -> None:
+        """Get remote and add climate from Switcher device."""
+        if coordinator.data.device_type.category == DeviceCategory.THERMOSTAT:
+            remote: BreezeRemote = await hass.async_add_executor_job(
+                remote_manager.get_remote, coordinator.data.remote_id
+            )
+            async_add_entities([SwitcherClimateEntity(coordinator, remote)])
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, SIGNAL_DEVICE_ADD, async_add_climate)
+    )
+
+
+class SwitcherClimateEntity(
+    CoordinatorEntity[SwitcherDataUpdateCoordinator], ClimateEntity
+):
+    """Representation of a Switcher climate entity."""
+
+    def __init__(
+        self, coordinator: SwitcherDataUpdateCoordinator, remote: BreezeRemote
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._remote = remote
+
+        self._attr_name = coordinator.name
+        self._attr_unique_id = f"{coordinator.device_id}-{coordinator.mac_address}"
```

Comment:
Can the MAC address be changed? If not, wouldn't that by itself be sufficient?

Reply:
MAC can't be changed, however historically the switch platform used both of them, so when adding new platforms I am trying to keep the same format, although I agree it is not needed.